### DISCUSSION TITLE
How do I make my custom SVGs for Badges?

### DISCUSSION THREAD

**Discussion by AbdullahAlhariri** [badges/shields#6882] [General] [answered]
Hi, I'm a beginner and I am trying to make my custom badge with my SVG on my computer
Is there any way to have my local SVG on the online badge
thanks.

**Comment by calebcartwright** [comment, marked as answer]
hello :wave: it's not entirely clear what you're asking, but it _sounds_ like you'd like to use a custom logo in your badge.

If that's indeed the case, would point you to our [docs on working with custom logos](https://github.com/badges/shields/blob/master/doc/logos.md#custom-logos). The gist though is that you first need to take your local SVG and convert it to base64, and there's a number of free tools available (including online converters). You'll then use that base64 output as the value for your query string parameter

**Reply by AbdullahAlhariri** [reply]
Thank you for responding.

So I should paste the base64 string into my URL, Which can result in a long URL. 
I have found  [this](https://img.shields.io/badge/c++-%2300599C.svg?style=for-the-badge&logo=c%2B%2B&logoColor=white) URL that's way shorter, my question now is
how is it possible to make like those short URLs

**Short URL**
https://img.shields.io/badge/c++-%2300599C.svg?style=for-the-badge&logo=c%2B%2B&logoColor=white

**Base64 URL**
https://img.shields.io/badge/play-station-blue.svg?logo=data:image/svg%2bxml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHZlcnNpb249IjEiIHdpZHRoPSI2MDAiIGhlaWdodD0iNjAwIj48cGF0aCBkPSJNMTI5IDExMWMtNTUgNC05MyA2Ni05MyA3OEwwIDM5OGMtMiA3MCAzNiA5MiA2OSA5MWgxYzc5IDAgODctNTcgMTMwLTEyOGgyMDFjNDMgNzEgNTAgMTI4IDEyOSAxMjhoMWMzMyAxIDcxLTIxIDY5LTkxbC0zNi0yMDljMC0xMi00MC03OC05OC03OGgtMTBjLTYzIDAtOTIgMzUtOTIgNDJIMjM2YzAtNy0yOS00Mi05Mi00MmgtMTV6IiBmaWxsPSIjZmZmIi8+PC9zdmc+

**Reply by calebcartwright** [reply]
Those are two completely different things.

Your "short" url is not using a custom logo, it is using one of the Simple Icon logos. This is also documented on the same doc page sent previously, https://github.com/badges/shields/blob/master/doc/logos.md#using-logos. If the Simple Icon set contains a logo you want, then you just need to specify the corresponding icon slug as-defined by Simple Icons.

If you want to use a custom logo as you've indicated, then you have to base64 encode the value. The Shields servers do not have access to your personal file system :wink: so if you want to use _your_ custom logo then you've got to provide it to Shields.

Yes, base64 strings are not known for their brevity, but a "short" url isn't the goal; the goal/feature here is allowing users to include _any_ logo/icon they want. Badge URLs are not something you have to type manually/deal with on a regular basis. You generate your url, and then stick it wherever you want your badge to display (e.g. Readme file) and then you forget about it

**Reply by AbdullahAlhariri** [reply]
Thanks for responding. 

I understand the goal of the base64 URLs. 
I wanted to give people a customizable badge directly from the URL
like the previous URL that I have provided to you.
Are there some Shields docs available for this purpose? (uploading SVG's for reuse in URLs)

Thanks a lot

**Reply by calebcartwright** [reply]
If your goal is to make your logo available for other people, you could consider providing custom [Endpoint badge](https://shields.io/endpoint) depending on your desired degree of customization. The Endpoint can set attributes like the logo (no base64 encoding required), but like our other dynamic badges, it doesn't allow for the message text to be configured.

However, custom logo distribution/marketing is not a goal/objective of the Endpoint badge. More generally the Shields project isn't in the business of promoting/marketing any product or icon, so if your ultimate goal is indeed to get other people to use your custom logo then you'll want to promote your logo elsewhere (and can include the base64 string in your docs to make it easier for people to add to their badges)

**Reply by AbdullahAlhariri** [reply]
Thanks for your awesome help

**Reply by AbdullahAlhariri** [reply]
> Your "short" url is not using a custom logo, it is using one of the Simple Icon logos. This is also documented on the same doc page sent previously, https://github.com/badges/shields/blob/master/doc/logos.md#using-logos. **If the Simple Icon set contains a logo you want, then you just need to specify the corresponding icon slug as-defined by Simple Icons.**

Hi, can you please provide me with the Simple Icon set. 

btw, thank you very much for your awesome work ♥

**Reply by AbdullahAlhariri** [reply]
> > Your "short" url is not using a custom logo, it is using one of the Simple Icon logos. This is also documented on the same doc page sent previously, https://github.com/badges/shields/blob/master/doc/logos.md#using-logos. **If the Simple Icon set contains a logo you want, then you just need to specify the corresponding icon slug as-defined by Simple Icons.**
> 
> Hi, can you please provide me with the Simple Icon set.
> 
> btw, thank you very much for your awesome work ♥

I just found it, thanks

**Reply by DevHanza** [reply]
Docs are also available on here: https://shields.io/docs/logos#custom-logos
